### PR TITLE
Spelling basic

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift basic, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

File: include/swift/Basic/Range.h [hunk 1]
```diff
@@ -134,7 +134,7 @@ class IntRange {
       return *this;
     }
     iterator operator+(difference_type i) const {
-      return iterator(Traits::adddOfset(Value, i));
+      return iterator(Traits::adddOffset(Value, i));
```
Comment (jsoref) [inline include/swift/Basic/Range.h:137]: ??

File: unittests/Basic/PrefixMapTest.cpp [hunk 2]
```diff
@@ -79,7 +79,7 @@ struct Tester {
     EXPECT_EQ(key.begin() + stdmapResult->first.size(), premapResult.second);
   }
 
-  // Perform a clear operation.  Tests that that actually clears out the map.
+  // Perform a clear operation.  Tests that the operation actually clears out the map.
```
Comment (jsoref) [inline unittests/Basic/PrefixMapTest.cpp:82]: This is a change to avoid doubled words. I'm happy to drop.

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: include/swift/Basic/Range.h (hunk 1 above)
Comment (CodaFi) [reply include/swift/Basic/Range.h:137]: This should be `Traits::addOffset`. This template member is not being instantiated so we never noticed the error...

**Comment by tshortli** [comment]
@swift-ci please smoke test Windows

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Review by tshortli — CHANGES_REQUESTED** [review]
Looks like the `adddOffset` -> `addOffset` feedback need to be addressed, but otherwise this looks good to go!

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Pull request merged**